astropy/wcs/src/wcslib_wrap.c

```diff
@@ -1622,6 +1622,10 @@ PyWcsprm_cset(
 
   int status = 0;
 
+  // We want to avoid calling wcsset whenever possible as it is not thread-safe. We use wcsenq
+  // to see if the checksum of the wcsprm elements has changed since wcsset was last called.
+  if (wcsenq(&self->x, WCSENQ_CHK)) return 0;
```

Comment:
Would it be worthwhile to add a comment that `wcsenq` itself is also not thread-safe, and therefore, should only be invoked while holding the GIL (i.e., not within a threaded region)?

Reply:
Just out of curiosity, which part of wcsenq is not thread-safe?

Reply:
Actually I was totally wrong! I thought `chksum` was the variable in the `wcsprm` struct but it is not. `chksum` is a function local variable and therefore, all threads should compute the same value of chksum in `wcsenq`